server/service/campaigns.go

```diff
@@ -81,7 +81,8 @@ func (svc *Service) NewDistributedQueryCampaign(ctx context.Context, queryString
 		}
 	}
 
-	if err := svc.authz.Authorize(ctx, query, fleet.ActionRun); err != nil {
+	tq := &fleet.TargetedQuery{Query: query, HostTargets: targets}
+	if err := svc.authz.Authorize(ctx, tq, fleet.ActionRun); err != nil {
```

Comment:
What about the `fleet.ActionRunNew` case?

Reply:
What do you mean? At this point the query is created, so the `ActionRun` access is validated. `ActionRunNew` check is above (line 66), when the query needs to be created.

Reply:
Ah, makes sense, thanks :)